<a href="https://colab.research.google.com/github/Emanuel2234/Analisis_causas_de_muerte/blob/main/An%C3%A1lisis_de_datos_sobre_Causas_de_muerte.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [15]:
#se importan las librerias
import pandas as pd
import numpy as np

In [16]:
#se importa el dataset
df = pd.read_csv('/content/cause_of_deaths.csv')

# Fase 1: entendimiento del negocio

In [17]:
#identicamos las columnas
df.columns

Index(['Country/Territory', 'Code', 'Year', 'Meningitis',
       'Alzheimer's Disease and Other Dementias', 'Parkinson's Disease',
       'Nutritional Deficiencies', 'Malaria', 'Drowning',
       'Interpersonal Violence', 'Maternal Disorders', 'HIV/AIDS',
       'Drug Use Disorders', 'Tuberculosis', 'Cardiovascular Diseases',
       'Lower Respiratory Infections', 'Neonatal Disorders',
       'Alcohol Use Disorders', 'Self-harm', 'Exposure to Forces of Nature',
       'Diarrheal Diseases', 'Environmental Heat and Cold Exposure',
       'Neoplasms', 'Conflict and Terrorism', 'Diabetes Mellitus',
       'Chronic Kidney Disease', 'Poisonings', 'Protein-Energy Malnutrition',
       'Road Injuries', 'Chronic Respiratory Diseases',
       'Cirrhosis and Other Chronic Liver Diseases', 'Digestive Diseases',
       'Fire, Heat, and Hot Substances', 'Acute Hepatitis'],
      dtype='object')

In [25]:
df.shape

(6120, 34)

**Se logra identificar loas 6120 registros en filas y las 34 columnas que serian las causas de muerte**


In [18]:
df.head(31)  #estos datos se basan desde el año 1990 hasta el 2019 para cada pais

,Country/Territory,Code,Year,Meningitis,Alzheimer's Disease and Other Dementias,Parkinson's Disease,Nutritional Deficiencies,Malaria,Drowning,Interpersonal Violence,...,Diabetes Mellitus,Chronic Kidney Disease,Poisonings,Protein-Energy Malnutrition,Road Injuries,Chronic Respiratory Diseases,Cirrhosis and Other Chronic Liver Diseases,Digestive Diseases,"Fire, Heat, and Hot Substances",Acute Hepatitis
0,Afghanistan,AFG,1990,2159,1116,371,2087,93,1370,1538,...,2108,3709,338,2054,4154,5945,2673,5005,323,2985
1,Afghanistan,AFG,1991,2218,1136,374,2153,189,1391,2001,...,2120,3724,351,2119,4472,6050,2728,5120,332,3092
2,Afghanistan,AFG,1992,2475,1162,378,2441,239,1514,2299,...,2153,3776,386,2404,5106,6223,2830,5335,360,3325
3,Afghanistan,AFG,1993,2812,1187,384,2837,108,1687,2589,...,2195,3862,425,2797,5681,6445,2943,5568,396,3601
4,Afghanistan,AFG,1994,3027,1211,391,3081,211,1809,2849,...,2231,3932,451,3038,6001,6664,3027,5739,420,3816
5,Afghanistan,AFG,1995,3102,1225,394,3131,175,1881,2969,...,2248,3974,467,3087,6211,6823,3076,5843,434,3946
6,Afghanistan,AFG,1996,3193,1239,398,3175,175,1969,3331,...,2298,4041,485,3130,6446,7000,3133,5966,450,4098
7,Afghanistan,AFG,1997,3304,1253,402,3250,240,2078,3028,...,2347,4094,504,3203,6677,7168,3200,6096,469,4252
8,Afghanistan,AFG,1998,3281,1267,405,3193,563,2098,3098,...,2388,4120,505,3145,6683,7287,3231,6154,472,4310
9,Afghanistan,AFG,1999,3200,1281,409,3115,468,2084,2917,...,2437,4150,502,3068,6603,7400,3247,6182,470,4332


##**Se logra apreciar que la columna "Country/Territory" y "Code" son las columnas categoricas para el resto de variables son numericas**


In [19]:
df['Country/Territory'].value_counts() #claramente se cuentan 30 para cada pais porque se repite por cada año los cuales son entre 1990 y 2019(se muestra la categoria y distribucion)

,count
Country/Territory,
Afghanistan,30
Albania,30
Algeria,30
American Samoa,30
Andorra,30
...,...
Venezuela,30
Vietnam,30
Yemen,30


In [26]:
df['Country/Territory'].value_counts().count()#muestra la cantidad

np.int64(204)

In [27]:
df['Code'].value_counts()

,count
Code,
AFG,30
ALB,30
DZA,30
ASM,30
AND,30
...,...
VEN,30
VNM,30
YEM,30


In [28]:
df['Code'].value_counts().count()

np.int64(204)

In [29]:
df.isna().sum() #no pose nulos

,0
Country/Territory,0
Code,0
Year,0
Meningitis,0
Alzheimer's Disease and Other Dementias,0
Parkinson's Disease,0
Nutritional Deficiencies,0
Malaria,0
Drowning,0
Interpersonal Violence,0


In [22]:
df.isnull().sum() #no posee valores nulos

,0
Country/Territory,0
Code,0
Year,0
Meningitis,0
Alzheimer's Disease and Other Dementias,0
Parkinson's Disease,0
Nutritional Deficiencies,0
Malaria,0
Drowning,0
Interpersonal Violence,0


In [23]:
df.describe().round() # estadistica a columnas numericas


,Year,Meningitis,Alzheimer's Disease and Other Dementias,Parkinson's Disease,Nutritional Deficiencies,Malaria,Drowning,Interpersonal Violence,Maternal Disorders,HIV/AIDS,...,Diabetes Mellitus,Chronic Kidney Disease,Poisonings,Protein-Energy Malnutrition,Road Injuries,Chronic Respiratory Diseases,Cirrhosis and Other Chronic Liver Diseases,Digestive Diseases,"Fire, Heat, and Hot Substances",Acute Hepatitis
count,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0,...,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0,6120.0
mean,2004.0,1720.0,4864.0,1173.0,2254.0,4141.0,1683.0,2084.0,1263.0,5942.0,...,5139.0,4724.0,425.0,1966.0,5931.0,17092.0,6124.0,10725.0,589.0,618.0
std,9.0,6672.0,18221.0,4616.0,10484.0,18428.0,8877.0,6917.0,6058.0,21012.0,...,16773.0,16470.0,2023.0,8256.0,24098.0,105157.0,20688.0,37228.0,2129.0,4186.0
min,1990.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,1.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0
25%,1997.0,15.0,90.0,27.0,9.0,0.0,34.0,40.0,5.0,11.0,...,236.0,146.0,6.0,5.0,175.0,289.0,154.0,284.0,17.0,2.0
50%,2004.0,109.0,666.0,164.0,119.0,0.0,177.0,265.0,54.0,136.0,...,1087.0,822.0,52.0,92.0,966.0,1689.0,1210.0,2185.0,126.0,15.0
75%,2012.0,847.0,2456.0,609.0,1167.0,393.0,698.0,877.0,734.0,1879.0,...,2954.0,2922.0,254.0,1042.0,3435.0,5250.0,3547.0,6080.0,450.0,160.0
max,2019.0,98358.0,320715.0,76990.0,268223.0,280604.0,153773.0,69640.0,107929.0,305491.0,...,273089.0,222922.0,30883.0,202241.0,329237.0,1366039.0,270037.0,464914.0,25876.0,64305.0


**El dataset tiene 6120 registros por variable donde se evaluaron desde los años 1990 hasta el 2019**


In [ ]:
df.dropna() #se eliminan nulos

In [ ]:
df_limpio = df.dropna() #aun no aplicado

In [ ]:
df_limpio.to_csv("/content/causas_muerte_limpio.csv") #solo ejecutar al finalizar el limpieza de datos

In [33]:
df.groupby("Year")["Malaria"].sum().sort_values(ascending=False)

,Malaria
Year,
2004,960784
2003,959654
2006,941746
2005,940540
2007,940193
2008,932910
2002,929195
2009,917796
2001,912709


In [20]:
df.groupby("Country/Territory")["Malaria"].sum().sort_values(ascending=False)

,Malaria
Country/Territory,
Nigeria,6422063
Democratic Republic of Congo,2557219
India,2439244
Uganda,1265629
Burkina Faso,950762
...,...
United Arab Emirates,0
Ukraine,0
Tuvalu,0


##**Se logra ver que los 5 paises con mas casos de muerte por malaria son Nigeria, Republica del Congo, India, Uganda y Burkina Faso**

In [21]:
df.groupby("Country/Territory")["Chronic Respiratory Diseases"].sum().sort_values(ascending=False)

,Chronic Respiratory Diseases
Country/Territory,
China,36676826
India,25232974
United States,4949052
Indonesia,2559457
Pakistan,2276479
...,...
San Marino,229
Cook Islands,213
Nauru,128


##**La mayor muerte por casos respiratorios estan entre china y la india superando el millon de muertos**